src/helpers/serverModuleHelpers.js

```diff
@@ -0,0 +1,122 @@
+const { existsSync } = require('node:fs')
+const { readFile, writeFile, rename, rm } = require('node:fs/promises')
+const { parse, join } = require('node:path')
+
+const { satisfies } = require('semver')
+
+const getAngularJson = require('./getAngularJson')
+const { getProject } = require('./setUpEdgeFunction')
+
+// eslint-disable-next-line no-inline-comments
+const NetlifyServerTsCommonEngine = /* typescript */ `import { CommonEngine } from '@angular/ssr/node'
```

Comment:
Obviously not in this PR, is anything blocking us moving to TS for our runtime code?

Reply:
> Obviously not in this PR, is anything blocking us moving to TS for our runtime code?

the `/* typescript */` ~pragma is just something that treats content of following string literal in compatible editors that support it - for stuff like this - https://marketplace.visualstudio.com/items?itemName=bierner.comment-tagged-templates

It doesn't have any effect otherwise.

As for moving entire code of angular runtime to typescript - this is completely separate and I don't there is blocker per se - but it would require a build step until stuff like https://nodejs.org/en/learn/typescript/run-natively is stable which I think is completely out of scope for this. I'm not sure how much benefit we would see here because angular runtime itself is not overly complex and it uses jsdoc already - maybe in between solution would be to use `// @ts-check` to enable typechecking and ensure that jsdocs are indeed correct and everything is type safe ( https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html ), but still not in scope for this PR I would say